[Xavier Ong] iP

xav168 opened the pull request:
# **Duke-san**

> "Discard everything that does not spark joy." - Marie Kondo [source](https://www.yourtango.com/2019322442/best-marie-kondo-quotes-inspire-de-clutter-life)

Duke-san is able to help you to do the following:

- Make a ToDo List
- Set a reminder for a deadline
- Has a cool interface

Here is how you can use it :

1. download it!
2. run it!
3. add your tasks
4. _profit!_ 🤑 

Features:

- [x] Supports DateTime format
- [x] Manage Tasks
- [ ] GUI (Coming Soon)

If you are a Java Programmer you can use it to practice java too! Here is the `main` method
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

xav168 committed:
Level 1 done

xav168 committed:
Level-2 Done

xav168 committed:
Level-4 done

xav168 committed:
Revert "Level-4 done"

This reverts commit 3ac10eeae9b145339e847f39d1d2dfc9e9437b43.

xav168 committed:
Level-3 Done

xav168 committed:
Level-4 Done

xav168 committed:
Implemented TextUiTesting

xav168 committed:
A-Exceptions

xav168 committed:
Level-6 Done

xav168 committed:
Added Enums

xav168 committed:
Level-7 done

xav168 committed:
Split classes into their own files

xav168 committed:
added DateTime formatting

xav168 committed:
Merge branch 'branch-Level-7'

xav168 committed:
Merge branch 'branch-Level-8'

xav168 committed:
Added More OOP

xav168 committed:
Added Packages

xav168 committed:
Added gradle support

xav168 committed:
Added JUnit Testing

xav168 committed:
added a Jar File

xav168 committed:
Added Javadocs

xav168 committed:
Fix case indentations

xav168 committed:
Add Find function

xav168 committed:
Merge branch 'branch-A-CodingStandard'

xav168 committed:
"Merge branch 'branch-A-JavaDoc'

xav168 committed:
Add checkstyle for gradle

xav168 committed:
Fix styling error according to checkstyle

xav168 committed:
Add config files for CheckStyle

mohamedsaf1 commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    /**
+     * Converts Deadline Object to a string that will be stored in a file
```
JavaDocs coding standard recommends a period (".") after sentences.

xav168 committed:
Add GUI for Duke

xav168 committed:
change gradle main function

xav168 committed:
Merge branch 'branch-A-CheckStyle'

xav168 committed:
fix style

connlim commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,85 @@
+    private ImageView displayPicture;
+
+    /**
+     * Constructor for DialogBox
+     * @param l
+     * @param iv
```
Java coding standard recommends empty line between description and parameter section. You should change it here and in all your other javadoc comments too.

connlim commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,59 @@
+ */
+public class Storage {
+    private Path path;
+    private File f;
```
```suggestion
    private File file;
```
Try to use more descriptive variable names for variables with large scopes (e.g. instance vars)

connlim commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+     * @return  condensed String represenation of Deadline Object
+     */
+    @Override
+    public String toFile() {
```
Unclear method name. `toFile` seems to mean that it will return a `File` object but it actually returns a `String` that can be used for storage. Perhaps try something like `toFileRepresentation`?

connlim commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,187 @@
+     *
+     * @param index
+     */
+    public String unMark(int index) {
```
```suggestion
     public String unmark(int index) {
```
'unmark' is a single word so the 'M' should not be capitalized.

connlim commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,105 @@
+    /**
+     * List of commands that the user can enter
+     */
+    enum Command {
+        list,
+        bye,
+        mark,
+        unmark,
+        delete,
+        deadline,
+        todo,
+        event,
+        find
+    }
```
Good use of enums!

xav168 committed:
Add assert in TaskList methods

There are instances where the user might pass in values that are greater
that the size of the list. This cases would not be ideal as it would
result in an error.

Using assert statements would ensure the value passed into the method
are valid.

xav168 committed:
Change names of variables

There are several variables with single letter names or names which did
not fit in to the Java "Code Quality" standards.

Changing those variable names adds clarity and readability to the code.

Let's try to find other areas in which we can improve our code quality.

xav168 committed:
Merge pull request #2 from xav168/branch-A-CodeQuality

Change names of variables

xav168 committed:
Merge branch 'master' into branch-A-Assertions

xav168 committed:
Merge pull request #1 from xav168/branch-A-Assertions

Add assert in TaskList methods

xav168 committed:
add remind functionality

xav168 committed:
Merge pull request #3 from xav168/branch-Reminders

add remind functionality

xav168 committed:
improve gui

xav168 committed:
fix jar creation

xav168 committed:
add more testing

xav168 committed:
Merge pull request #5 from xav168/branch-A-MoreTesting

add more testing

xav168 committed:
fix some style to adhere to coding standards

xav168 committed:
fix style for tests

xav168 committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

xav168 committed:
add Welcome message and shuts down after bye command

xav168 committed:
change text formatting to look cleaner

xav168 committed:
Add Ui.png

xav168 committed:
Update README.md

xav168 committed:
Update to reduce length of methods

xav168 committed:
Fix Event class
